app/src/ui/lib/list/section-list.tsx

```diff
@@ -0,0 +1,1680 @@
+  private focusRow: RowIndexPath = InvalidRowIndexPath
+
+  private readonly rowRefs = new RowRefsMap()
+
+  /**
+   * The style prop for our child Grid. We keep this here in order
+   * to not create a new object on each render and thus forcing
+   * the Grid to re-render even though nothing has changed.
+   */
+  private gridStyle: React.CSSProperties = { overflowX: 'hidden' }
+
+  /**
+   * On Win32 we use a fake scroll bar. This variable keeps track of
+   * which of the actual scroll container and the fake scroll container
+   * received the scroll event first to avoid bouncing back and forth
+   * causing jerky scroll bars and more importantly making the mouse
+   * wheel scroll speed appear different when scrolling over the
+   * fake scroll bar and the actual one.
+   */
+  private lastScroll: 'grid' | 'fake' | null = null
+
+  private list: HTMLDivElement | null = null
+  private rootGrid: Grid | null = null
+  private grids = new Map<number, Grid>()
+  private readonly resizeObserver: ResizeObserver | null = null
+  private updateSizeTimeoutId: NodeJS.Immediate | null = null
+
+  /**
+   * Get the props for the inner scroll container (called containerProps on the
+   * Grid component). This is memoized to avoid causing the Grid component to
+   * rerender every time the list component rerenders (the Grid component is a
+   * pure component so a complex object like containerProps being instantiated
+   * on each render would cause it to rerender constantly).
+   */
+  private getContainerProps = memoizeOne(
+    (
+      activeDescendant: string | undefined
+    ): React.HTMLProps<HTMLDivElement> => ({
+      onKeyDown: this.onKeyDown,
+      'aria-activedescendant': activeDescendant,
+      'aria-multiselectable':
+        this.props.selectionMode === 'multi' ||
+        this.props.selectionMode === 'range'
+          ? 'true'
+          : undefined,
+    })
+  )
+
+  public constructor(props: ISectionListProps) {
+    super(props)
+
+    this.state = {
+      scrollTop: 0,
+    }
+
+    const ResizeObserverClass: typeof ResizeObserver = (window as any)
+      .ResizeObserver
+
+    if (ResizeObserver || false) {
+      this.resizeObserver = new ResizeObserverClass(entries => {
+        for (const { target, contentRect } of entries) {
+          if (target === this.list && this.list !== null) {
+            // We might end up causing a recursive update by updating the state
+            // when we're reacting to a resize so we'll defer it until after
+            // react is done with this frame.
+            if (this.updateSizeTimeoutId !== null) {
+              clearImmediate(this.updateSizeTimeoutId)
+            }
+
+            this.updateSizeTimeoutId = setImmediate(
+              this.onResized,
+              this.list,
+              contentRect
+            )
+          }
+        }
+      })
+    }
+  }
+
+  private get totalRowCount() {
+    return this.props.rowCount.reduce((a, b) => a + b, 0)
```

Comment:
Nit.. because this method is so small.. But, you have an exportable `getTotalRowCount` in your `list-row-index-path` that could be reused here.